core/src/main/java/jenkins/security/apitoken/ApiTokenStore.java

```diff
@@ -371,6 +371,22 @@ private void init() {
             
             return result;
         }
+    
+        public static @Nonnull HashedToken buildNewFromLegacy(@Nonnull HashValue value, boolean migrationFromExistingLegacy) {
```

Comment:
I already packed it away, so I'm not sure which item it is, but Effective Java suggests using enums instead of booleans for situations like this. The context of what exactly that flag does is lost in a language like Java (i.e., no passing parameters by name).

Reply:
In this case the method is `@Restricted` and so can be considered "private" for us, i.e. the context is never lost as we are using the source directly.